services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidateSimpleFlowSpec.groovy

```diff
@@ -0,0 +1,517 @@
+""")
+class MetersValidateSimpleFlowSpec extends BaseSpecification {
+    @Autowired
+    SwitchHelper switchHelper
+
+    def "Able to get meter validate information"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def defaultMetersSrcSwitch = northbound.getAllMeters(srcSwitch.dpId)
+        def defaultMetersDstSwitch = northbound.getAllMeters(dstSwitch.dpId)
+        def flow = flowHelper.addFlow(flowHelper.randomFlow(srcSwitch, dstSwitch))
+
+        then: "One meter is automatically created on the both switches"
+        def createdMetersIdSrcSwitch = getCreatedMetersId(srcSwitch.dpId, defaultMetersSrcSwitch)
+        def createdMetersIdDstSwitch = getCreatedMetersId(dstSwitch.dpId, defaultMetersDstSwitch)
+        def burstSizeSrcSwitch = switchHelper.getExpectedBurst(srcSwitch.dpId, flow.maximumBandwidth)
+        def burstSizeDstSwitch = switchHelper.getExpectedBurst(dstSwitch.dpId, flow.maximumBandwidth)
+
+        createdMetersIdSrcSwitch.size() == 1
+        createdMetersIdDstSwitch.size() == 1
+
+        and: "The correct info is stored in the validate meter response"
+        def meterValidateInfoSrcSwitch = northbound.validateMeters(srcSwitch.dpId)
+        def meterValidateInfoDstSwitch = northbound.validateMeters(dstSwitch.dpId)
+        def createdCookieSrcSwitch = getCreatedCookies(srcSwitch.dpId, createdMetersIdSrcSwitch)
+        def createdCookieDstSwitch = getCreatedCookies(dstSwitch.dpId, createdMetersIdDstSwitch)
+
+        def createdCookies = createdCookieSrcSwitch + createdCookieDstSwitch
+
+        meterValidateInfoSrcSwitch.meters.proper.collect { it.meterId }.containsAll(createdMetersIdSrcSwitch)
+        meterValidateInfoDstSwitch.meters.proper.collect { it.meterId }.containsAll(createdMetersIdDstSwitch)
+
+        meterValidateInfoSrcSwitch.meters.proper.collect { it.cookie }.containsAll(createdCookieSrcSwitch)
+        meterValidateInfoDstSwitch.meters.proper.collect { it.cookie }.containsAll(createdCookieDstSwitch)
+
+        [meterValidateInfoSrcSwitch, meterValidateInfoDstSwitch].each {
+            it.meters.proper.each {
+                assert it.rate == flow.maximumBandwidth
+                assert it.flowId == flow.id
+            }
+        }
+
+        meterValidateInfoSrcSwitch.meters.proper.burstSize[0] == burstSizeSrcSwitch
+        meterValidateInfoDstSwitch.meters.proper.burstSize[0] == burstSizeDstSwitch
+
+        [meterValidateInfoSrcSwitch, meterValidateInfoDstSwitch].each {
+            it.meters.proper.every {
+                ["KBPS", "BURST", "STATS"].containsAll(it.flags)
+            }
+        }
+
+        [srcSwitch, dstSwitch].each {
+            checkNoMeterValidation(it.dpId, ["proper"])
+        }
+
+        and: "Created rules are stored in the proper section"
+        [meterValidateInfoSrcSwitch, meterValidateInfoDstSwitch].each {
+            it.rules.properRules.containsAll(createdCookies)
```

Comment:
Please, if possible, reduce the amount of code between point of declaration and point of use of the `createdCookies` variable (and other vars if possible). It is declared for no reason at L48 and used only at L78. This makes it hard for the reader to get into the context